src/test/java/org/corfudb/runtime/view/LayoutViewTest.java

```diff
@@ -70,4 +71,45 @@ public void canTolerateLayoutServerFailure()
 
         r.getStreamsView().get(CorfuRuntime.getStreamID("hi")).check();
     }
+
+    @Test
+    public void canAvoidNullException()
+            throws Exception {
+        CorfuRuntime r = getDefaultRuntime().connect();
+        Layout l = new TestLayoutBuilder()
+                .setEpoch(1L)
+                .addLayoutServer(9000)
+                .addSequencer(9000)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(9000)
+                .addToSegment()
+                .addToLayout()
+                .build();
+        l.setRuntime(r);
+        r.getLayoutView().updateLayout(l, 1L);
+        r.invalidateLayout();
+
+        Layout l2 = new TestLayoutBuilder()
+                .setEpoch(2L)
+                .addLayoutServer(9000)
+                .addSequencer(9000)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(9000)
+                .addToSegment()
+                .addToLayout()
+                .build();
+        l2.setRuntime(r);
+        r.getLayoutView().updateLayout(l2, 1L);
+
+        try {
```

Comment:
This test logic is not clear. 

Are you asserting that, if updateLayout(l1, 1L) is called and a WrongEpochException is thrown, that it should have a WrongEpochException with correct epoch field of 2?

Or, are you asserting that calling updateLayout should throw a WrongEpochException, with a correct epoch of 2? 

If the latter is the case, you should use assertThatThrownBy or some other way to describe the intent - see:
http://joel-costigliola.github.io/assertj/assertj-core-news.html#assertj-core-3.0.0-thrown-by